### ISSUE TITLE
Hyprland crashes when switching KVM sources

### ISSUE THREAD

**Issue by rykugur** [hyprwm/Hyprland#6756]
### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 918d8340afd652b011b937d29d5eea0be08467f5  (flake.lock: update).
Date: 2024-06-25
Tag: v0.41.2, commits: 918d8340afd652b011b937d29d5eea0be08467f5

flags: (if any)


System Information:
System name: Linux
Node name: jezrien
Release: 6.9.7
Version: #1-NixOS SMP PREEMPT_DYNAMIC Thu Jun 27 11:52:32 UTC 2024


GPU information: 
2f:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 32 [Radeon RX 7700 XT / 7800 XT] [1002:747e] (rev c8) (prog-if 00 [VGA controller])


os-release: ANSI_COLOR="1;34"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="24.11.20240629.2741b4b"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
IMAGE_ID=""
IMAGE_VERSION=""
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 24.11 (Vicuna)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="24.11 (Vicuna)"
VERSION_CODENAME=vicuna
VERSION_ID="24.11"


plugins:

======Config-Start======


Config File: /home/dusty/.config/hypr/hyprland.conf: Read Succeeded
exec-once = /nix/store/y29cbwfiwb9nyi4db4apk9dgylzdjm1v-dbus-1.14.10/bin/dbus-update-activation-environment --systemd DISPLAY HYPRLAND_INSTANCE_SIGNATURE WAYLAND_DISPLAY XDG_CURRENT_DESKTOP && systemctl --user stop hyprland-session.target && systemctl --user start hyprland-session.target
source = ~/.dotfiles/hosts/jezrien/hyprland.conf
source = ~/.dotfiles/configs/hypr/default.conf
source = ~/.dotfiles/configs/hypr/binds.conf
source = ~/.dotfiles/configs/hypr/input.conf
source = ~/.dotfiles/configs/hypr/rules.conf


Config File: /home/dusty/.dotfiles/hosts/jezrien/hyprland.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Monitors/
# Rotations: monitor = ...,transform,#
# normal (no transforms) -> 0
# 90 degrees -> 1
# 180 degrees -> 2
# 270 degrees -> 3
# flipped -> 4
# flipped + 90 degrees -> 5
# flipped + 180 degrees -> 6
# flipped + 270 degrees -> 7
monitor=DP-1,3440x1440@144,0x1440,1
monitor=DP-2,3440x1440@144,0x0,1
# monitor=HDMI-A-1,1920x1080@144,3440x0,1,transform,1

# exec-once = sleep 1 && waybar
# this causes problems if started on login
# exec-once = steam
exec-once = 1password
exec-once = discord
exec-once = dunst
exec-once = wl-paste --watch cliphist store
exec-once = solaar -w hide
exec-once = easyeffects --gapplication-service
exec-once = playerctld daemon

misc {
  vrr = 2
}

debug {
    disable_logs = false
  }

workspace = 1,monitor:DP-1
workspace = 2,monitor:DP-1
workspace = 3,monitor:DP-1
workspace = 4,monitor:DP-1
workspace = 5,monitor:DP-1
# workspace = 6,monitor:HDMI-A-1,layoutopt:orientation:top
# workspace = 7,monitor:HDMI-A-1,layoutopt:orientation:top
workspace = 6,monitor:DP-2
workspace = 7,monitor:DP-2


Config File: /home/dusty/.dotfiles/configs/hypr/default.conf: Read Succeeded
# set some env variables
env = XDG_SESSION_TYPE,wayland
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_DESKTOP,Hyprland

env = QT_QPA_PLATFORM,wayland
env = QT_QPA_PLATFORMTHEME,qt5ct # change to qt6ct if you have that

env = COPYCMD,wl-copy
env = PASTECMD,wl-paste

env = XCURSOR_SIZE,24

$terminal = kitty
$fileManager = nemo
$menu = fuzzel

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 5
    gaps_out = 20
    border_size = 2
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)

    layout = dwindle

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10
    
    blur {
        enabled = true
        size = 3
        passes = 1
    }

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = true
    first_launch_animation = true

    bezier = wind, 0.05, 0.9, 0.1, 1.05
    bezier = winIn, 0.1, 1.1, 0.1, 1.1
    bezier = winOut, 0.3, -0.3, 0, 1
    bezier = liner, 1, 1, 1, 1
    animation = windows, 1, 6, wind, slide
    animation = windowsIn, 1, 6, winIn, slide
    animation = windowsOut, 1, 5, winOut, slide
    animation = windowsMove, 1, 5, wind, slide
    animation = border, 1, 1, liner
    animation = borderangle, 1, 30, liner, loop
    animation = fade, 1, 10, default
    animation = workspaces, 1, 5, wind
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
    force_split = 2
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    # new_is_master = false
}


Config File: /home/dusty/.dotfiles/configs/hypr/binds.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
# TODO: replace these binds
bind = $mainMod, Return, exec, $terminal
# bind = $mainMod, Return, exec, wezterm
bind = $mainMod, C, killactive, 
# bind = $mainMod SHIFT, E, exit
bind = $mainMod SHIFT, E, exec, wlogout
# bind = $mainMod, M, exit, 
# bind = $mainMod, E, exec, $fileManager
bind = $mainMod, F, fullscreen
bind = $mainMod, V, togglefloating 
bind = $mainMod, R, exec, $menu
bind = $mainMod, P, pseudo, # dwindle
# bind = $mainMod, J, togglesplit, # dwindle

bind = $mainMod, h, movefocus, l
bind = $mainMod, j, movefocus, d
bind = $mainMod, k, movefocus, u
bind = $mainMod, l, movefocus, r

bind = $mainMod SHIFT, h, movewindow, l
bind = $mainMod SHIFT, j, movewindow, d
bind = $mainMod SHIFT, k, movewindow, u
bind = $mainMod SHIFT, l, movewindow, r 

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7

# Example special workspace (scratchpad)
bind = $mainMod, 0, togglespecialworkspace, magic
bind = $mainMod SHIFT, 0, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

# Media keys
binde = , XF86AudioRaiseVolume, exec, amixer sset Master 5%+
binde = , XF86AudioLowerVolume, exec, amixer sset Master 5%-
bind = , XF86AudioMute, exec, amixer sset Master toggle
bind = , XF86AudioPlay, exec, playerctl play-pause
bind = , XF86AudioPause, exec, playerctl play-pause
bind = , XF86AudioNext, exec, playerctl next
bind = , XF86AudioPrev, exec, playerctl previous
bind = , XF86MonBrightnessUp, exec, xbacklight +5 && notify-send "Brightness - $(xbacklight -get | cut -d '.' -f 1)%"
bind = , XF86MonBrightnessDown, exec, xbacklight -5 && notify-send "Brightness - $(xbacklight -get | cut -d '.' -f 1)%"

# Misc binds
bind = $mainMod SHIFT, F1, exec, hyprprop | jq '.class' | wl-copy
bind = $mainMod SHIFT, Print, exec, grim -g "$(slurp)" - | swappy -f -
bind = $mainMod SHIFT, S, exec, grim -g "$(slurp)" - | wl-copy
bind = $mainMod SHIFT, B, exec, ~/.config/waybar/launch.fish
bind = $mainMod SHIFT, V, exec, cliphist list | wofi --show dmenu


Config File: /home/dusty/.dotfiles/configs/hypr/input.conf: Read Succeeded
input {
    kb_layout = us
    kb_variant =
    kb_model =
    kb_options =
    kb_rules =

    follow_mouse = 2
    # fixes oddities with drop-down menu items, e.g. steam
    mouse_refocus = false

    touchpad {
        natural_scroll = no
        tap-to-click = true
        clickfinger_behavior = true
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.

    repeat_rate = 60
    repeat_delay = 200
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = true
    workspace_swipe_fingers = 3
    workspace_swipe_invert = false
}



Config File: /home/dusty/.dotfiles/configs/hypr/rules.conf: Read Succeeded
# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more

windowrulev2 = float, class:(1Password)
windowrulev2 = float, class:(com.obsproject.Studio)
windowrulev2 = float, class:(galculator)
windowrulev2 = float, class:(neovide)
windowrulev2 = float, class:(nemo)
windowrulev2 = float, class:(pavucontrol)
windowrulev2 = float, class:(ristretto) # image viewer
windowrulev2 = float, class:(yad)
windowrulev2 = float, class:(virt-manager)
windowrulev2 = float, class:(vlc)
windowrulev2 = float, class:(zenity)

windowrulev2 = workspace 1, class:(firefox)
windowrulev2 = workspace 1, class:(Google-chrome)

windowrulev2 = workspace 4 silent, class:(steam)
windowrulev2 = workspace 4 silent, title:(Friends List)
windowrulev2 = float, class:(steam), title:(Friends List)
windowrulev2 = stayfocused, title:^()$,class:^(steam)$
windowrulev2 = minsize 1 1, title:^()$,class:^(steam)$

windowrulev2 = workspace 4 silent, class:(discord)

# dota
windowrulev2 = workspace 5, class:(dota2)
# project zomboid
windowrulev2 = workspace 5, class:(Project Zomboid)
windowrulev2 = fullscreen, class:(Project Zomboid)
# X4
windowrulev2 = workspace 5, class:(X4)

======Config-End========


```

</details>

### Description

NOTE: I marked this as "Yes" for regression, but I'm not sure on that. I just added this KVM last week. I'm also seeing intermittent crashes that _seem_ to be related to waking from suspend and/or dpms, which others have already reported, and I'm unsure if there's any connection there.

Intermittently, switching between KVM sources will cause a Hyprland crash. I'm not too well versed in reading these logs/crash logs, but nothing obvious was sticking out to me.

Just tested several times, waiting random amounts of time in between (10 seconds to 10 minutes); might be more related to something like dpms than I thought, but I'm not going to pretend to know a ton about that.

## Setup

I have 2x 34" monitors (1440p) connected via DP (Desktop -> KVM -> Monitors) I also have a Macbook Pro connected via Macbook -> Dock -> KVM -> Monitors. Everything works fine.

### How to reproduce

Have KVM
Switch KVM sources away from Hyprland PC and back
Sometimes crash

### Crash reports, logs, images, videos

See this gist for logs and crash logs:

https://gist.github.com/rykugur/c063e9c01c1ef490536c001255492469

**Comment by gulafaran** [comment]
could you try produce a crash log with debug symbols not stripped out and preferrably even with asan?

**Comment by rykugur** [comment]
> could you try produce a crash log with debug symbols not stripped out and preferrably even with asan?

I'll get debug up and running this evening. I don't see an asan package in the Hyprland flake; will check into building it manually this evening as well.

**Comment by gulafaran** [comment]
oh its just a build options WITH_ASAN

**Comment by gulafaran** [comment]
but no idea how nixos deals with building things which i assume flake is heh

**Comment by vaxerski** [comment]
also try on git.

**Comment by rykugur** [comment]
I tried building git / asan but ran into a number of blockers that I didn't resolve.

I did however muck with AMDGPU power profiles today while trying to figure out why some games were flickering at random times, and _so far_ manually setting [power profiles](https://wiki.archlinux.org/title/AMDGPU#Power_profiles) seems to have also resolved this issue.

I also recently updated my system, which may have also resolved it.

I'll keep an eye on it and will close this issue if it doesn't pop back up.
